### PR TITLE
Optimize and upgrade dependency versions

### PR DESCRIPTION
**Pull request by yihezkel**
#### Pull Request Description
Mostly transparent optimizations

---

#### Future Release Comment
**Features:**
- Optimize and upgrade dependency versions
- Minor code optimizations

**Fixes:**
- Resolve mvn warnings
- Remove dependency duplicates/conflicts

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,32 +35,33 @@
...
-        <httpcore.version>4.4.11</httpcore.version>
-        <json.version>20190722</json.version>
-        <fasterxml.jackson.core.version>[2.11.0,2.12.0)</fasterxml.jackson.core.version>
-        <azure-storage.version>8.3.0</azure-storage.version>
+        <commons-lang3.version>3.11</commons-lang3.version>
+        <msal4j.version>1.9.1</msal4j.version>
+        <httpclient.version>4.5.13</httpclient.version>
+        <httpcore.version>4.4.13</httpcore.version>
+        <json.version>20201115</json.version>
+        <fasterxml.jackson.core.version>2.12.1</fasterxml.jackson.core.version>
```
Comment (yihezkel) [inline pom.xml:45]: Previously was [2.11.0,2.12.0) - why?

File: ingest/pom.xml [hunk 2]
```diff
@@ -88,8 +88,6 @@
                     <ignoredUnusedDeclaredDependencies>
                         <dependency>org.xerial:sqlite-jdbc</dependency>
                         <dependency>org.slf4j:slf4j-simple</dependency>
-                        <dependency>org.junit.jupiter:junit-jupiter-engine</dependency>
```
Comment (yihezkel) [inline ingest/pom.xml:91]: Why do we need this? The first is already scoped for tests only, and if the second is just for test scope why not scope for tests only?

File: data/pom.xml [hunk 3]
```diff
@@ -94,9 +94,6 @@
                 <version>${maven-dependency-plugin.version}</version>
                 <configuration>
                     <failOnWarning>true</failOnWarning>
-                    <ignoredUnusedDeclaredDependencies>
-                        <dependency>org.junit.jupiter:junit-jupiter-engine</dependency>
```
Comment (yihezkel) [inline data/pom.xml:98]: Why do we need this? It's already scoped for tests only

File: pom.xml (hunk 1 above)
Comment (ohadbitt) [reply pom.xml:45]: i dont know but i do know that all these jackson libraries are a big mess of versions

**Review by ohadbitt — APPROVED** [review]

**Pull request merged**